src/server/modes/charts/plugins/ql/utils/migrate-helpers.ts

```diff
@@ -3,100 +3,142 @@ import type {QlConfigResultEntryMetadataDataColumnOrGroup} from '../../../../../
 
 import {isGroup} from './misc-helpers';
 
+const findIndexes = ({fields, forbiddenIndexes}: {fields: Field[]; forbiddenIndexes: number[]}) =>
+    fields
+        .map((field, fieldIndex) =>
+            (field.cast === DATASET_FIELD_TYPES.INTEGER ||
+                field.cast === DATASET_FIELD_TYPES.FLOAT) &&
+            !forbiddenIndexes.includes(fieldIndex)
+                ? fieldIndex
+                : null,
+        )
+        .filter((value) => value !== null) as number[];
```

Comment:
to avoid going through the array twice, you can use reduce

Reply:
Fixed